chpl/chpl-service/src/main/java/gov/healthit/chpl/activity/history/explorer/RealWorldTestingEligibilityActivityExplorer.java

```diff
@@ -30,31 +33,36 @@ public RealWorldTestingEligibilityActivityExplorer(ActivityDAO activityDao) {
 
     @Override
     @Transactional
-    public ActivityDTO getActivity(ListingActivityQuery query) {
+    public List<ActivityDTO> getActivities(ListingActivityQuery query) {
         ActivityDTO activityReturn = null;
         RealWorldTestingEligibilityQuery realWorldTestingEligibilityQuery = (RealWorldTestingEligibilityQuery) query;
-        List<ActivityDTO> listingActivities = getAllActivityforListing(realWorldTestingEligibilityQuery.getCertifiedProductId());
+        List<ActivityDTO> listingActivities = getAllActivityforListing(realWorldTestingEligibilityQuery.getListingId());
 
         if (realWorldTestingEligibilityQuery.getAsOfDate() == null) {
-            if (listingActivities.size() > 0) {
+            if (!CollectionUtils.isEmpty(listingActivities)) {
                 activityReturn = listingActivities.get(0);
             }
         } else {
             Iterator<ActivityDTO> listingActivityIter = listingActivities.iterator();
             while (listingActivityIter.hasNext()) {
                 ActivityDTO currActivity = listingActivityIter.next();
-
                 if (currActivity.getActivityDate().before(new Date(DateUtil.toEpochMillis(realWorldTestingEligibilityQuery.getAsOfDate())))) {
                     activityReturn = currActivity;
                 }
             }
         }
-        return activityReturn;
+
+        if (activityReturn == null) {
+            return Collections.emptyList();
+        }
+        return Stream.of(activityReturn).toList();
```

Comment:
Should we regression test whatever this RWT explorer used this call for? It looks like there might be multiple activities for a given day (time period?) and we should make sure that didn't break anything, right?

Reply:
Feel free to look at the test cases. I made notes about these changes during my development and included regression tests for all of them.